v03_pipeline/lib/tasks/reference_data/updated_reference_dataset_collection_test.py

```diff
@@ -0,0 +1,106 @@
+import shutil
+from unittest import mock
+from unittest.mock import ANY
+
+import hail as hl
+import luigi.worker
+
+from v03_pipeline.lib.model import (
+    DatasetType,
+    ReferenceDatasetCollection,
+    ReferenceGenome,
+    SampleType,
+)
+from v03_pipeline.lib.tasks.reference_data.updated_reference_dataset_collection import (
+    UpdatedReferenceDatasetCollectionTask,
+)
+from v03_pipeline.lib.test.mocked_dataroot_testcase import MockedDatarootTestCase
+
+COMBINED_2_PATH = 'v03_pipeline/var/test/reference_data/test_combined_2.ht'
+PRIMATE_AI_TEST_PATH = 'v03_pipeline/var/test/reference_data/primate_ai_test.ht'
+
+
+class UpdatedReferenceDatasetCollectionTaskTest(MockedDatarootTestCase):
+    @mock.patch(
+        'v03_pipeline.lib.tasks.reference_data.updated_reference_dataset_collection.valid_reference_dataset_collection_path',
+    )
+    @mock.patch(
+        'v03_pipeline.lib.reference_data.dataset_table_operations.get_dataset_ht',
+    )
+    @mock.patch.object(ReferenceDatasetCollection, 'datasets')
+    def test_update_with_dataset(
+        self,
+        mock_reference_dataset_collection_datasets,
+        mock_get_dataset_ht,
+        mock_destination_path,
+    ) -> None:
+        mock_reference_dataset_collection_datasets.return_value = ['primate_ai']
+        # delete the dataset collection directory to fail initial task 'complete()' and force 'run()'
+        # since collection won't exist, we can't test the codepath for updating an existing collection yet
+        shutil.rmtree(COMBINED_2_PATH)
+        mock_destination_path.return_value = COMBINED_2_PATH
```

Comment:
One tiny thing here... the existing pattern for this has been to move the in-repo hts to the tmp dir setup by the `MockedDatarootTestCase`.  [here](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/tasks/update_variant_annotations_table_with_new_samples_test.py#L79-L86) is an example.

The goal there has been to use a unique tmpdir for every test so that (one day) they're fully parallelizable.  We also probably don't want to just mock `valid_reference_dataset_collection_path` since it's gonna get called multiple times with multiple different arguments under the hood!

You can use moving the file as a proxy for "has the task run yet" in the way you sorta are now.  There's an existing example [here](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/tasks/update_variant_annotations_table_with_new_samples_test.py#L138-L159).

Reply:
thanks for linking examples, helped me figure out the missing shutil  piece. I was close!